kedro-datasets/kedro_datasets/constants.py

```diff
@@ -0,0 +1,16 @@
+"""
+`constants.py` defines custom data types for Kedro-viz integration. It uses NewType from the typing module.
+These types are used to facilitate data rendering in the Kedro-viz front-end.
+"""
+
+from typing import NewType
+
+Dataframe = NewType("Dataframe", dict)
+Image = NewType("Image", str)
+Plot = NewType("Plot", dict)
+JSON = NewType("JSON", str)
```

Comment:
Can you explain why `Dataframe` is a subclass of `dict` here? I am not too familiar with this `typing`, why do we choose `NewType` over `TypeVar`?